lib/blacklight/path.rb

```diff
@@ -0,0 +1,131 @@
+module Blacklight
+  # This class encapsulates the search state as represented by the query
+  # parameters namely: :f, :q, :page, :per_page and, :sort
+  class Path
+    include Blacklight::Facet
```

Comment:
Is this just for `facet_configuration_for_field`? I wonder if we should push that up into `Blacklight::Configuration` (now, or in a future PR)